.github/workflows/functional.yml

```diff
@@ -0,0 +1,157 @@
+name: Functional Testing
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+  pull_request:
+  workflow_call:
```

Comment:
I don't think this trigger is needed. Just push/pull request.

Reply:
update: we need this trigger but only for image to avoid lint errors, image is being used for getting secrets during docker build of NGINX

Reply:
I don't understand. This workflow should build images in the exact same way as the `conformance` workflow. That workflow doesn't need this input, so why would this one? What linting are you referring to?

Reply:
I thought it needed the secrets because I was running into cert and key issues, but looks like that is not the case. I have removed it.